src/Controller/Hello.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class Hello
+{
+    use ControllerTrait;
+
+    /**
+     * @Route("/", name="homepage")
+     */
+    public function __invoke(Request $request): Response
```

Comment:
@stof thanks for the explanation, to be clearer, we are defined two possible controllers here (at the same time):

 1. `'App\Controller\Hello'`
 2. `'App\Controller\Hello::__invoke'`

Currently we are using the second controller with the appearance to use the first one, so my suggestion here is to use a better method name if we use already `@route()` annotation, otherwise let's define the controller into some `routing.yml` to use the first one only.

Now in Symfony it is not possible to declare this type of route as annotation but could be good to have something like this:
```
/**
 * @Route("/", name="homepage")
 */
class Hello
{
    public function __invoke()
    {
        return new Response('Hello World!');
    }
}
```

Actually equal to: 

```
homepage:
    path: /
    defaults: { _controller: src\Controller\Hello }
```